app/src/engineering/java/com/hedvig/app/feature/offer/MockOfferViewModel.kt

```diff
@@ -61,41 +74,66 @@ class MockOfferViewModel : OfferViewModel() {
-            val documentItems = OfferItemsBuilder.createDocumentItems(mockData.quoteBundle.quotes)
-            val insurableLimitsItems = OfferItemsBuilder.createInsurableLimits(mockData.quoteBundle.quotes)
-            val bottomOfferItems = OfferItemsBuilder.createBottomOfferItems(mockData)
-            _viewState.value =
-                ViewState(
-                    topOfferItems = topOfferItems,
-                    perils = perilItems,
-                    documents = documentItems,
-                    insurableLimitsItems = insurableLimitsItems,
-                    bottomOfferItems = bottomOfferItems,
-                    signMethod = mockData.signMethodForQuotes,
-                    checkoutLabel = mockData.checkoutLabel(),
-                    title = mockData.quoteBundle.appConfiguration.title,
-                    loginStatus = LoginStatus.LOGGED_IN
+            do {
+                if (shouldError) {
+                    _viewState.value = ViewState.Error
+                    return@launch
+                }
+                val topOfferItems = OfferItemsBuilder.createTopOfferItems(
+                    mockData.offer,
+                    mockData.dataCollectionValue,
+                    mockData.externalInsuranceData
                 )
+                val perilItems = OfferItemsBuilder.createPerilItems(mockData.offer.quoteBundle.quotes)
+                val documentItems = OfferItemsBuilder.createDocumentItems(mockData.offer.quoteBundle.quotes)
+                val insurableLimitsItems = OfferItemsBuilder.createInsurableLimits(mockData.offer.quoteBundle.quotes)
+                val bottomOfferItems = OfferItemsBuilder.createBottomOfferItems(mockData.offer)
+                _viewState.value =
+                    ViewState.Content(
+                        topOfferItems = topOfferItems,
+                        perils = perilItems,
+                        documents = documentItems,
+                        insurableLimitsItems = insurableLimitsItems,
+                        bottomOfferItems = bottomOfferItems,
+                        signMethod = mockData.offer.signMethodForQuotes,
+                        checkoutLabel = mockData.offer.checkoutLabel(),
+                        title = mockData.offer.quoteBundle.appConfiguration.title,
+                        loginStatus = LoginStatus.LOGGED_IN
+                    )
+                delay(2000)
+            } while (mockRefreshEvery2Seconds)
         }
     }
 
     companion object {
         var shouldError = false
-        var mockData = OFFER_DATA_SWEDISH_APARTMENT
+        var mockData: OfferMockData = OfferMockData(OFFER_DATA_SWEDISH_APARTMENT)
+        var mockRefreshEvery2Seconds = false
+
+        data class OfferMockData(
+            val offer: OfferQuery.Data,
+            val dataCollectionValue: SubscribeToDataCollectionUseCase.Status? = null,
+            val externalInsuranceData: DataCollectionResultQuery.Data? = null,
+        ) {
```

Comment:
Since the data now come from 3 different queries, just bunched them up in a class